packages/react/src/analytics/integrations/Vitorino/providers/Riskified/Riskified.ts

```diff
@@ -0,0 +1,149 @@
+   *
+   * @param options - Custom options for the integration.
+   * @param loadData - Analytics's load event data.
+   */
+  constructor(
+    options: RiskifiedIntegrationOptions,
+    loadData: LoadIntegrationEventData,
+    analytics: StrippedDownAnalytics,
+  ) {
+    super(options, loadData, analytics);
+
+    this.isReady = false;
+    this.siteId = options.siteId || 'farfetch.com';
+    this.baseUrl = `https://beacon.riskified.com?shop=${this.siteId}&sid=`;
+
+    this.setupScript(loadData);
+  }
+
+  /* static createInstance<Options extends IntegrationOptions>(
+    this: IntegrationFactory<Options>,
+    options: Options,
+*/
+  static override createInstance<Options extends RiskifiedIntegrationOptions>(
+    this: IntegrationFactory<Options>,
+    options: Options,
+    loadData: LoadIntegrationEventData,
+    analytics: StrippedDownAnalytics,
+  ) {
+    return new this(options, loadData, analytics);
+  }
+
+  /**
+   * Returns true due to being a required integration - No need to check for consent.
+   *
+   * @returns A value indicating if the integration should load.
+   */
+  static override shouldLoad(): boolean {
+    return true;
+  }
+
+  /**
+   * Creates the script based on the correlation `id` and appends it on the page.
+   * Once the script is on the page, it will automatically call `RISKX.go()`.
+   *
+   * @param loadData - Analytics's load event data.
+   *
+   * @returns The Riskified instance to allow chaining calls.
+   */
+  setupScript(loadData: LoadIntegrationEventData): this {
+    const correlationId = get(loadData, 'user.localId');
+    const script = document.createElement('script');
+    const url = `${this.baseUrl}${correlationId}`;
+
+    document.body.appendChild(script);
+
+    script.onload = this.onLoad;
+
+    script.setAttribute('data-test', 'riskified');
+    script.async = true;
+    script.src = url;
+
+    return this;
+  }
+
+  /**
+   * Tracks events (page views or page actions).
+   *
+   * @param data - The event object.
+   *
+   * @returns The Riskified instance to allow chaining calls.
+   */
+  override track(data: EventData<TrackTypesValues>) {
+    switch (data.type) {
+      case TrackTypes.PAGE:
+        return this.trackPage(get(data, 'context.web.window.location.href'));
+
+      case TrackTypes.TRACK:
+        return null;
+
+      default:
+        return null;
+    }
+  }
+
+  /**
+   * Tracks the page view event. Calls the global Riskified variable RISKX.go with the web location.
+   *
+   * @param location - The location data.
+   *
+   * @returns The Riskified instance to allow chaining calls.
+   */
+  trackPage(location: string): this {
+    if (!this.isReady) {
+      return this;
+    }
+
+    window.RISKX.go(location);
+
+    return this;
+  }
+
+  /**
+   * When the script loads onto the DOM, it will set the integration as ready.
+   */
+  onLoad = () => {
+    this.isReady = true;
+  };
+
+  getOptions() {
```

Comment:
is there a use-case for these getters here?

Reply:
Removed and refactored the tests